pkg/receive/handler.go

```diff
@@ -91,3 +103,49 @@ func (h *Handler) Receive(w http.ResponseWriter, r *http.Request) {
 	h.forwardRequestsTotal.WithLabelValues("success").Inc()
 	w.WriteHeader(resp.StatusCode)
 }
+
+var ErrRequiredLabelMissing = fmt.Errorf("a required label is missing from the metric")
+
+// ValidateLabels makes sure that the request's content contains the required partitionKey
+func (h *Handler) ValidateLabels(next http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		bodyBytes, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, "failed to read body", http.StatusInternalServerError)
+			return
+		}
+		r.Body.Close()
+
+		r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+		body, err := ioutil.ReadAll(r.Body)
+
+		content, err := snappy.Decode(nil, body)
+		if err != nil {
+			http.Error(w, "failed to decode request body", http.StatusBadRequest)
+			return
+		}
+
+		var wreq prompb.WriteRequest
+		if err := proto.Unmarshal(content, &wreq); err != nil {
+			http.Error(w, "failed to decode protobuf from body", http.StatusBadRequest)
+			return
+		}
+
+		found := false
+		for _, ts := range wreq.GetTimeseries() {
+			for _, l := range ts.GetLabels() {
+				if l.Name == h.PartitionKey {
```

Comment:
I think this logic is wrong. This currently checks if at least one time series on the request contains the label. We want to ensure that every time series has that label.